pkg/oc/admin/router/router.go

```diff
@@ -655,6 +659,7 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 		"STATS_PORT":                            strconv.Itoa(cfg.StatsPort),
 		"STATS_USERNAME":                        cfg.StatsUsername,
 		"STATS_PASSWORD":                        cfg.StatsPassword,
+		"ROUTER_THREADS":                        strconv.Itoa(cfg.Threads),
```

Comment:
Would prefer we not set this here.  I don't see why a user would set this flag, and not just expect us to do the right thing.

Reply:
What is the right number for the threads?  How do we know what else they want to use the machine for?  Or are you suggesting we pick it up from the container limits?  The other problem there is that the number of processes that are running will vary if they have long-running connections and have router reloads.

I really don't want to enable this by default yet, so we at least need a way to disable it.  I'm fine with an "auto" setting for num threads, but I suspect we will need a way to tune it.

Reply:
+1 on auto setting the default num threads based on pod cpu limits and allow this value to be overridden via env var or some other mechanism.

Reply:
Without knowing how effective our auto-detected number is in practice, does it make any sense to add an explicitly unsupported parameter to give an escape hatch and allow us to test tunings without a new build?

When we have higher confidence our auto-detected algo is solid we could talk about removing the override. There is a bit of UX complexity there in explaining the mutual exclusivity or precedence order of the auto/explicit options, but if the override is considered unsupported/debugging-only maybe it isn't as big an issue?